src/framework/components/sound/component.js

```diff
@@ -13,160 +15,107 @@ import { SoundSlot } from './slot.js';
-    this.on("set_maxDistance", this.onSetMaxDistance, this);
-    this.on("set_rollOffFactor", this.onSetRollOffFactor, this);
-    this.on("set_distanceModel", this.onSetDistanceModel, this);
-    this.on("set_positional", this.onSetPositional, this);
+    this._volume = 1;
+    this._pitch = 1;
+    this._positional = true;
+    this._refDistance = 1;
+    this._maxDistance = 10000;
+    this._rollOffFactor = 1;
+    this._distanceModel = DISTANCE_LINEAR;
+    this._slots = {};
+
+    this._playingBeforeDisable = {};
 }
 SoundComponent.prototype = Object.create(Component.prototype);
 SoundComponent.prototype.constructor = SoundComponent;
 
-Object.assign(SoundComponent.prototype, {
-    onSetSlots: function (name, oldValue, newValue) {
-        var key;
-
-        // stop previous slots
-        if (oldValue) {
-            for (key in oldValue) {
-                oldValue[key].stop();
-            }
-        }
-
-        var slots = {};
-
-        // convert data to slots
-        for (key in newValue) {
-            if (!(newValue[key] instanceof SoundSlot)) {
-                if (newValue[key].name) {
-                    slots[newValue[key].name] = new SoundSlot(this, newValue[key].name, newValue[key]);
-                }
-            } else {
-                slots[newValue[key].name] = newValue[key];
-            }
-        }
-
-        this.data.slots = slots;
-
-        // call onEnable in order to start autoPlay slots
-        if (this.enabled && this.entity.enabled)
-            this.onEnable();
-    },
-
-    onSetVolume: function (name, oldValue, newValue) {
-        var slots = this.data.slots;
-        for (var key in slots) {
-            var slot = slots[key];
-            // change volume of non-overlapping instances
-            if (!slot.overlap) {
-                var instances = slot.instances;
-                for (var i = 0, len = instances.length; i < len; i++) {
-                    instances[i].volume = slot.volume * newValue;
-                }
-            }
-        }
-    },
+function defineSoundPropertyBasic(publicName, privateName) {
+    Object.defineProperty(SoundComponent.prototype, publicName, {
+        get: function () {
+            return this[privateName];
+        },
+        set: function (newValue) {
+            this[privateName] = newValue;
 
-    onSetPitch: function (name, oldValue, newValue) {
-        var slots = this.data.slots;
-        for (var key in slots) {
-            var slot = slots[key];
-            // change pitch of non-overlapping instances
-            if (!slot.overlap) {
-                var instances = slot.instances;
-                for (var i = 0, len = instances.length; i < len; i++) {
-                    instances[i].pitch = slot.pitch * newValue;
+            var slots = this._slots;
+            for (var key in slots) {
+                var slot = slots[key];
+                // change refDistance of non-overlapping instances
```

Comment:
refDistance not relevant to this comment

Reply:
Fixed.